exca/task.py

```diff
@@ -457,3 +470,136 @@ def __init__(self, infra: TaskInfra) -> None:
     def __call__(self) -> tp.Any:
         # this method replaces the decorated method
         return self.infra._infra_method()  # type: ignore
+
+
+## similar to TaskInfra but without cache
+
+
+class SubmitInfra(base.BaseInfra, slurm.SubmititMixin):
+    """Processing infrastructure ready to be applied to a pydantic.BaseModel method.
+    To use it, the configuration must be set as an attribute of a pydantic BaseModel,
+    then :code:`@infra.apply` must be set on the method to process
+    this will effectively replace the function with a remotely-computed version of itself.
+    Contrarily to TaskInfra, outputs of the method are not cached, and the method can take arguments
+
+    Parameters
+    ----------
+    folder: optional Path or str
+        Path to directory for dumping/loading the cache on disk, if provided
+
+    Slurm/submitit parameters
+    -------------------------
+    Check out :class:`exca.slurm.SubmititMixin`
+
+    Usage
+    -----
+    .. code-block:: python
+
+        class MyTask(pydantic.BaseModel):
+            x: int
+            infra: exca.TaskInfra = exca.TaskInfra()
+
+            @infra.apply
+            def compute(self, y: int) -> int:
+                return 2 * self.x
+
+        cfg = MyTask(12, infra={"folder": "tmp", "cluster": "slurm"})
+        assert cfg.compute(y=1) == 25  # "compute" runs on slurm but is not cached
+        job = cfg.infra.submit(y=1)  # runs the computation asynchronously
+        assert job.result() == 25
+
+    Note
+    ----
+    The decorated method can be a staticmethod to avoid pickling the owner object
+    along with the other parameters.
```

Comment:
```suggestion
    - The decorated method can be a staticmethod to avoid pickling the owner object
      along with the other parameters.
    - This is an experimental infra that is still evolving
```